x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -117,7 +118,20 @@ public class TransportGetStackTracesAction extends HandledTransportAction<GetSta
     private final int desiredSlices;
     private final int desiredDetailSlices;
     private final boolean realtime;
-    private static final Map<String, HostMetadata> hostsTable = new ConcurrentHashMap<>();
+
+    /**
+     * We just apply an upper limit to the number of cached hosts to ensure it doesn't blow up
+     * in unforeseen situations.
+     * The value itself is relatively arbitrary, but high enough to cover most foreseeable use cases.
+     */
+    private static final int MAX_HOSTS_CACHE_SIZE = 20_000;
+    private static final Map<String, HostMetadata> hostsTable =
```

Comment:
Can we extend the comment to describe the reasoning behind this cache? It's hard to immediately see how it plays into the CO2 cost calculator.

Reply:
I'll discuss with Daniel tomorrow whether we drop the global cache.
Though it's hard to reason about - our largest cluster just has a few hundred hosts ingesting. And that is so low-cardinality that querying that host metadata is fast enough to add 0 latency to the API calls (stacktraces/flamegraph). Since we officially only support up to 5k cores, the number of different hosts we support is roughly in the range of what our testing cluster receives.

Reply:
We decided to drop the cache as it creates more headaches than it curates.
This PR has been replaced by https://github.com/elastic/elasticsearch/pull/102687 .

Reply:
Thanks for bringing this up, @christos68k . While I knew about the issues with caching host metadata, at some point I decided to ignore this until "later" and implemented a very naive cache.

If not using a cache ever turns out to be an issue for customers, we likely need a design doc and go down the rabbit hole.